### PR TITLE
CFE-2817: Support @if macro on line 1 

### PR DESCRIPTION
**Pull request by basvandervlies**

### CODE REVIEW

File: libpromises/cf3lex.l [hunk 1]
```diff
@@ -140,14 +140,15 @@ promise_type   [a-zA-Z_]+:
...
+                          P.line_pos = 1;
                           // push back on stack and skip first char
                           yyless(1);
                       }
 
 {macro_if_version}    {
-                        if (  P.line_pos != 0 )
+                        if (  P.line_pos != 1 )
                         {
+                             printf("line %d, pos = %d\n", P.line_no, P.line_pos);
```
Comment (vpodzime) [inline libpromises/cf3lex.l]: Is ``printf()`` really a good thing to call here?

Comment (basvandervlies) [reply libpromises/cf3lex.l]: sorry is a left over.  Good catch.

**Comment by vpodzime** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/541/)

**Comment by nickanderson** [comment]
The build failure seemd to be unrelated.
On solaris 9

```
Error Message

FAIL (The test Passed, but failure was expected) (UNEXPECTED FAILURE) ./10_files/09_insert_lines/017.cf

Stacktrace

2018-05-14T20:14:35%z     info: filestat bailing on link "/perl5/5.6.1/bin/perl" (original "/bin/perl") because it could not be read
2018-05-14T20:14:36%z     info: filestat bailing on link "/perl5/5.6.1/bin/perl" (original "/bin/perl") because it could not be read
2018-05-14T20:14:37%z     info: filestat bailing on link "/perl5/5.6.1/bin/perl" (original "/bin/perl") because it could not be read
2018-05-14T20:14:39%z     info: filestat bailing on link "/perl5/5.6.1/bin/perl" (original "/bin/perl") because it could not be read
2018-05-14T20:14:40%z     info: filestat bailing on link "/perl5/5.6.1/bin/perl" (original "/bin/perl") because it could not be read
2018-05-14T20:14:41%z     info: Created directory "/home/jenkins/workspace/testing-pr/label/PACKAGES_sparc64_solaris_9/cfengine-3.13.0a.aaf3c4d/tests/acceptance/workdir/__10_files_09_insert_lines_017_cf/tmp/TESTDIR.cfengine/."
R: /home/jenkins/workspace/testing-pr/label/PACKAGES_sparc64_solaris_9/cfengine-3.13.0a.aaf3c4d/tests/acceptance/./10_files/09_insert_lines/../../dcs.cf.sub SFAIL/redmine6336
2018-05-14T20:14:41%z     info: Created file "/home/jenkins/workspace/testing-pr/label/PACKAGES_sparc64_solaris_9/cfengine-3.13.0a.aaf3c4d/tests/acceptance/workdir/__10_files_09_insert_lines_017_cf/tmp/TEST.cfengine", mode 0600
2018-05-14T20:14:41%z    error: Could not read file /home/jenkins/workspace/testing-pr/label/PACKAGES_sparc64_solaris_9/cfengine-3.13.0a.aaf3c4d/tests/acceptance/workdir/__10_files_09_insert_lines_017_cf/tmp: Is a directory
R: This should only pass if inserting lines from a directory fails
R: /home/jenkins/workspace/testing-pr/label/PACKAGES_sparc64_solaris_9/cfengine-3.13.0a.aaf3c4d/tests/acceptance/./10_files/09_insert_lines/017.cf Pass
        
```

**Comment by nickanderson** [comment]
This should be backported to 3.10.x and 3.7.x if possible

**Comment by basvandervlies** [comment]
@Nick and the rest of the crew. It is only a bug in master and 3.12, due the parsing changed to support long lines/hashes. I do not know which issue that is/was

**Comment by olehermanse** [comment]
@basvandervlies Thanks, found the old commit, that makes sense.

**Comment by olehermanse** [comment]
This has been cherry-picked in #3132

**Pull request merged**